python/pyspark/sql/functions.py

```diff
@@ -1848,6 +1848,12 @@ def __del__(self):
             self._broadcast = None
 
     def __call__(self, *cols):
+        for c in cols:
+            if not isinstance(c, (Column, str)):
+                raise TypeError(
+                    "All arguments should be Columns or strings representing column names. "
```

Comment:
Yes, it is. My point is it is not always obvious for new users. Giving a hint that there is such thing as literals could be a good idea.